pkg/configurator/crd_client.go

```diff
@@ -0,0 +1,182 @@
+}
+
+// Listens to ConfigMap events and notifies dispatcher to issue config updates to the envoys based
+// on config seen on the configmap
+func (c *CRDClient) startMeshConfigListener() error {
+	// Subscribe to configuration updates
+	cfgSubChannel := events.GetPubSubInstance().Subscribe(
+		announcements.MeshConfigAdded,
+		announcements.MeshConfigDeleted,
+		announcements.MeshConfigUpdated,
+	)
+
+	// run the listener
+	go func(cfgSubChannel chan interface{}, cf *CRDClient) {
+		for {
+			msg := <-cfgSubChannel
+
+			psubMsg, ok := msg.(events.PubSubMessage)
+			if !ok {
+				log.Error().Msgf("Could not cast pubsub message")
+				continue
+			}
+
+			switch psubMsg.AnnouncementType {
+			case announcements.MeshConfigAdded:
+				log.Debug().Msgf("[%s] OSM MeshConfig added event triggered a global proxy broadcast",
+					psubMsg.AnnouncementType)
+				events.GetPubSubInstance().Publish(events.PubSubMessage{
+					AnnouncementType: announcements.ScheduleProxyBroadcast,
+					OldObj:           nil,
+					NewObj:           nil,
+				})
+
+			case announcements.MeshConfigDeleted:
+				// Ignore deletion. We expect config to be present
+				log.Debug().Msgf("[%s] OSM MeshConfig deleted event triggered a global proxy broadcast",
+					psubMsg.AnnouncementType)
+				events.GetPubSubInstance().Publish(events.PubSubMessage{
+					AnnouncementType: announcements.ScheduleProxyBroadcast,
+					OldObj:           nil,
+					NewObj:           nil,
+				})
+
+			case announcements.MeshConfigUpdated:
+				// Get config map
+				prevMeshConfigObj, okPrevCast := psubMsg.OldObj.(*v1alpha1.MeshConfig)
+				newMeshConfigObj, okNewCast := psubMsg.NewObj.(*v1alpha1.MeshConfig)
+				if !okPrevCast || !okNewCast {
+					log.Error().Msgf("[%s] Error casting old/new MeshConfigs objects (%v %v)",
+						psubMsg.AnnouncementType, okPrevCast, okNewCast)
+					continue
+				}
+
+				// Parse old and new configs
+				prevMeshConfig := parseOSMMeshConfig(prevMeshConfigObj)
+				newMeshConfig := parseOSMMeshConfig(newMeshConfigObj)
+
+				// Determine if we should issue new global config update to all envoys
+				triggerGlobalBroadcast := false
+
+				triggerGlobalBroadcast = triggerGlobalBroadcast || (prevMeshConfig.Egress != newMeshConfig.Egress)
+				triggerGlobalBroadcast = triggerGlobalBroadcast || (prevMeshConfig.PermissiveTrafficPolicyMode != newMeshConfig.PermissiveTrafficPolicyMode)
+				triggerGlobalBroadcast = triggerGlobalBroadcast || (prevMeshConfig.UseHTTPSIngress != newMeshConfig.UseHTTPSIngress)
+				triggerGlobalBroadcast = triggerGlobalBroadcast || (prevMeshConfig.TracingEnable != newMeshConfig.TracingEnable)
+				triggerGlobalBroadcast = triggerGlobalBroadcast || (prevMeshConfig.TracingAddress != newMeshConfig.TracingAddress)
+				triggerGlobalBroadcast = triggerGlobalBroadcast || (prevMeshConfig.TracingEndpoint != newMeshConfig.TracingEndpoint)
+				triggerGlobalBroadcast = triggerGlobalBroadcast || (prevMeshConfig.TracingPort != newMeshConfig.TracingPort)
+
```

Comment:
prometheus_scrapping is missing I think, assuming it's also present in meshconfig crd

Reply:
It is removed on purpose. @shashankram  can you confirm this?

Reply:
We added it recenrly on ToT as it changing this value needs to reconfigure existing envoys to configure the lds listener for prometheus metrics (check pkg/envoy/lds/response,go@88), unless something change in LDS land (@shashankram ?)

Reply:
oh maybe you mean we will not be using this flag anymore in days to come, so entirely removed from cfg?

Reply:
@allenlsy was there a reason to remove this? Nothing has changed in LDS wrt the scraping config. @eduser25 perhaps this configuration will be voided by #3065 , but till then we should keep the event unless there is something I am missing.

Reply:
+1, we are on the same page. @allenlsy I think we will still need it for now.

Reply:
Will add it back